### PR TITLE
Add tests for shouldDelete function

### PR DESCRIPTION
**Pull request by nrb**
**What type of PR is this?**
> /kind failing-test

**What this PR does / why we need it**:
Adds more test coverage to the sidecar-controller package

**Which issue(s) this PR fixes**:
Part of #204 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by xing-yang** [comment]
Please choose a PR type and add NONE in the release notes.

**Comment by nrb** [comment]
Sorry for not filling in the template - I was trying the VSCode GitHub PR extension and didn't realize it would skip the PR template.

File: pkg/sidecar-controller/snapshot_controller_test.go [hunk 1]
```diff
@@ -89,3 +90,53 @@ func TestControllerCacheParsingError(t *testing.T) {
 		t.Errorf("Expected parsing error, got nil instead")
 	}
 }
+
+//TestShouldDelete tests logic for deleting VolumeSnapshotContent objects.
```
Comment (xing-yang) [inline pkg/sidecar-controller/snapshot_controller_test.go]: nit: space after //

File: pkg/sidecar-controller/snapshot_controller_test.go [hunk 2]
```diff
@@ -89,3 +90,53 @@ func TestControllerCacheParsingError(t *testing.T) {
...
+
+	tests := []struct {
+		name           string
+		expectedReturn bool
+		content        *crdv1.VolumeSnapshotContent
+	}{
+		{
+			name:           "DeletionTimeStamp is nil",
+			expectedReturn: false,
+			content:        &crdv1.VolumeSnapshotContent{},
```
Comment (xing-yang) [inline pkg/sidecar-controller/snapshot_controller_test.go]: can you populate the content with some fields

**Comment by nrb** [comment]
@xing-yang Addressed both comments

**Comment by xing-yang** [comment]
Looks good.  Can you squash the commits to 1?  Thanks.

**Comment by nrb** [comment]
Done!

**Comment by xing-yang** [comment]
Thanks @nrb!

/lgtm
/approve

**Pull request merged**